examples/rcn.py

```diff
@@ -0,0 +1,392 @@
+# ---
+# jupyter:
+#   jupytext:
+#     formats: ipynb,py:percent
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#       jupytext_version: 1.13.1
+#   kernelspec:
+#     display_name: Python 3
+#     language: python
+#     name: python3
+# ---
+
+# %%
+import os
+import time
+from typing import Dict
+
+import jax
+import matplotlib.pyplot as plt
+
+# %%
+# %matplotlib inline
+import numpy as np
+from jax import numpy as jnp
+from jax import tree_util
+from scipy.ndimage import maximum_filter
+from scipy.signal import fftconvolve
+from sklearn.datasets import fetch_openml
+
+from pgmax.fg import graph, groups
+
+os.environ["XLA_PYTHON_CLIENT_ALLOCATOR"] = "platform"
+os.environ["XLA_PYTHON_CLIENT_PREALLOCATE"] = "false"
+
+# %% [markdown]
+# # 1. Load the data
+# %%
+hps, vps = 12, 12
+train_size = 20
```

Comment:
In my test, on a GPU with 10 GB of memory, with https://github.com/vicariousinc/PGMax/pull/97 you are now able to run inference with 100 templates in one batch. I suggest we change `train_size` to 100 (so that the detections/performances would look better).

Reply:
Using `train_size = 100` has the following advantages. 
1. Backtraces might look better. 
2. Higher accuracy.

Disadvantages.
1. A person who does not have a powerful gpu will probably not be able to use it. 
2. Notebook takes a long time to finish on the CPU. For small train set, notebook finishes in a reasonable time even on the CPU. 

If the backtraces look decent on for the smaller case, might be better to use size = 20. Let me know what you think.

Reply:
Good point on the requirement of a powerful GPU. Maybe we can give user both options? So load all 100 models, and subset to use only 20 if needed (a simple indexing change), and add a comment detailing the considerations/trade-off.

Reply:
Added support for 100 templates but the default is still 20.
Can switch and wrote comments for explanation.